### PR TITLE
Adapts FeaturePerfCSCollection setup and adds synthetic case studies

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats/varats/projects/perf_tests/feature_perf_cs_collection.py [hunk 1]
```diff
@@ -19,10 +20,35 @@
...
 from varats.project.varats_project import VProject
+from varats.utils.git_commands import init_all_submodules, update_all_submodules
 from varats.utils.git_util import RevisionBinaryMap, ShortCommitHash
 from varats.utils.settings import bb_cfg
 
 
+def _do_feature_perf_cs_collection_compile(
+    project: VProject, cmake_flag: str
+) -> None:
+    """Base class that implements common project functionality."""
```
Comment (boehmseb) [inline varats/varats/projects/perf_tests/feature_perf_cs_collection.py]: That docstring doesn't really fit here.

File: varats/varats/projects/perf_tests/feature_perf_cs_collection.py [hunk 2]
```diff
@@ -120,18 +146,240 @@ def run_tests(self) -> None:
...
+
+    NAME = 'SynthSAFieldSensitivity'
+    GROUP = 'perf_tests'
+    DOMAIN = ProjectDomains.TEST
+
+    SOURCE = [
+        bb.source.Git(
+            remote="https://github.com/se-sic/FeaturePerfCSCollection.git",
+            local="SynthSAFieldSensitivity",
+            refspec="origin/f-StaticAnalysisMotivatedSynthBenchmarks",
```
Comment (boehmseb) [inline varats/varats/projects/perf_tests/feature_perf_cs_collection.py]: Before merging, I guess this should be changed to `origin/HEAD`.

**Review by boehmseb — APPROVED** [review]
Looks good overall. Only minor things to clean up.

File: varats/varats/projects/perf_tests/feature_perf_cs_collection.py (hunk 2 above)
Comment (vulder) [reply varats/varats/projects/perf_tests/feature_perf_cs_collection.py]: Yep, I just needed this to make everything testable on my system. After merging https://github.com/se-sic/FeaturePerfCSCollection/pull/11 this is now reverted.

**Pull request merged**